makefile

```diff
@@ -11,7 +11,7 @@ ifeq ($(UNAME), Darwin) # macOS
 	LDFLAGS += -L/usr/local/include -lboost_system -lboost_filesystem
 endif
 
-DEBUG_FLAGS = -g -Wall -Werror
+DEBUG_FLAGS = -g -Wall -Wextra -Werror
```

Comment:
Personally, I'm not a huge fan of `Wextra` since a lot of its warnings are for unused variables and parameters and macros, which may or may not be used in normal places. I'm fine with leaving it on, but instead of changing all the functions to bend around backwards for the compiler flags, maybe we can selectively disable the ones that are most annoying? I usually do something like `WARNINGS += -Wno-old-style-cast -Wno-reserved-id-macro -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-newline-eof -Wno-documentation -Wno-documentation-unknown-command -Wno-unused-function -Wno-unused-parameter -Wno-unused-macros`